Sources/TuistSupport/Utils/DerivedDataLocator.swift

```diff
@@ -4,23 +4,52 @@ import Path
 
 public protocol DerivedDataLocating {
     func locate(
-        for projectPath: AbsolutePath
+        workspacePath: AbsolutePath,
+        schemeName: String
     ) throws -> AbsolutePath
 }
 
 public final class DerivedDataLocator: DerivedDataLocating {
-    private let devEnvironment: DeveloperEnvironmenting
+    private let fileHandler: FileHandling
 
-    public init(devEnvironment: DeveloperEnvironmenting = DeveloperEnvironment.shared) {
-        self.devEnvironment = devEnvironment
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
     }
 
     public func locate(
-        for projectPath: AbsolutePath
+        workspacePath: AbsolutePath,
+        schemeName: String
     ) throws -> AbsolutePath {
-        let hash = try XcodeProjectPathHasher.hashString(for: projectPath.pathString)
-        return devEnvironment.derivedDataDirectory
-            .appending(component: "\(projectPath.basenameWithoutExt)-\(hash)")
+        let hash = try XcodeProjectPathHasher.hashString(for: workspacePath.pathString)
+        return try derivedDataDirectory(
+            workspacePath: workspacePath,
+            schemeName: schemeName
+        ).appending(component: "\(workspacePath.basenameWithoutExt)-\(hash)")
+    }
+
+    private func derivedDataDirectory(
+        workspacePath: AbsolutePath,
+        schemeName: String
+    ) throws -> AbsolutePath {
+        let location: AbsolutePath
+        if let currentLocation = try? System.shared.capture([
+            "/bin/bash",
+            "-c",
+            """
+            xcodebuild -workspace '\(workspacePath)' -scheme '\(schemeName)' -showBuildSettings | \
+                grep MODULE_CACHE_DIR | \
+                head -1 | \
+                cut -d '=' -f 2 | \
+                xargs
+            """,
```

Comment:
A couple of points around this:
- Let's move the implementation to `XcodeBuildController`
- I'd run this only in CI environments. If you use the vanilla SPM integration, this command resolves SPM packages and can be thus quite slow

That being said, I wonder if we should use this command at all. Relying on the scheme name is not ideal – I think I'd lean to using `IDEDerivedDataPathOverride` or reading the `CI_DERIVED_DATA_PATH` instead. Wdyt?

Reply:
I'd be fine with just using `IDEDerivedDataPathOverride`. It's a small change, and we could always use the command later if there were a more compelling reason.

Reply:
Sounds good, let's use `IDEDerivedDataPathOverride` as the risks attached to using it are lower.